src/main/java/org/jnode/fs/xfs/XfsObject.java

```diff
@@ -97,4 +123,72 @@ public long getUInt32(int relativeOffset) {
     public long getInt64(int relativeOffset) {
         return BigEndian.getInt64(data, offset + relativeOffset);
     }
+
+    public long read(long offset, int size) throws IOException {
```

Comment:
I would remove this function and replace where it's being called with the expected function eg. `read(0, 1)` with `getUInit8(0)`.
* It's clearer to see what's expected at the calling function.
* you don't need to worry about `RuntimeException`.

Reply:
code change applied